pandas/_libs/src/ujson/lib/ultrajsonenc.c

```diff
@@ -1080,6 +1087,39 @@ void encode(JSOBJ obj, JSONObjectEncoder *enc, const char *name,
         }
 
         case JT_UTF8: {
+            printf("case: JT_UTF8\n");
+            value = enc->getStringValue(obj, &tc, &szlen);
+            Buffer_Reserve(enc, RESERVE_STRING(szlen));
+            if (enc->errorMsg) {
+                enc->endTypeContext(obj, &tc);
+                return;
+            }
+            Buffer_AppendCharUnchecked(enc, '\"');
+
+            if (enc->forceASCII) {
+                if (!Buffer_EscapeStringValidated(obj, enc, value,
+                                                  value + szlen)) {
+                    enc->endTypeContext(obj, &tc);
+                    enc->level--;
+                    return;
+                }
+            } else {
+                if (!Buffer_EscapeStringUnvalidated(enc, value,
+                                                    value + szlen)) {
+                    enc->endTypeContext(obj, &tc);
+                    enc->level--;
+                    return;
+                }
+            }
+
+            Buffer_AppendCharUnchecked(enc, '\"');
+            break;
+        }
+
+        case JT_BIGNUM: {
```

Comment:
The implementation here should be pretty similar to the UTF8 case but remove the `Buffer_AppendCharUnchecked(enc, '\"');` lines. You also don't need the `enc->forceASCII` check and can just do `1Buffer_EscapeStringUnvalidated`

Reply:
I think you will also need to define a function pointer for a callback that converts the object to a string representation

See:

https://github.com/pandas-dev/pandas/blob/e79487d8e50a03be87188608cd1379330a4fd224/pandas/_libs/src/ujson/python/objToJSON.c#L2116

And 

https://github.com/pandas-dev/pandas/blob/e79487d8e50a03be87188608cd1379330a4fd224/pandas/_libs/src/ujson/python/objToJSON.c#L1659

As examples

Reply:
So this works up to line 1112, in that
``` C++
 value = enc->getBigNumStringValue(obj, &tc, &szlen);
```
gets the correct string (I checked with a print statement). But there's still a problem because we exit with an `OverflowError` via this line (checked with a print statement)
https://github.com/pandas-dev/pandas/blob/5d69786af46291abe727b290a84aa9d2663cd50c/pandas/_libs/src/ujson/python/objToJSON.c#L2289
I feel like this something to do with the buffer and the length of the string we're printing out somehow not being right...?

Reply:
This goes back to L1636. CPython manages a global error and we never explicitly clear it on that line. You could all `PyErr_Clear()` but it actually would be better if you used `PyLong_AsLongLongAndOverflow` and checked for a non-zero overflow

https://docs.python.org/3/c-api/long.html#c.PyLong_AsLongLongAndOverflow

Reply:
Ok! I'll go with `PyLong_AsLongLongAndOverflow`.

Quick question, though: when I do
``` C++
        int *_overflow = 0;
        GET_TC(tc)->longValue = PyLong_AsLongLongAndOverflow(obj, _overflow);
```
I'm making a mistake (there's a segfault) - but not sure why

Reply:
The way it is written you are telling the program to write the overflow result to memory location 0, which the program almost assuredly doesn't have access to (hence segfault)

Just declare the int and use the address of operator

```c
int overflow;
GET_TC(tc)->longValue = PyLong_AsLongLongAndOverflow(obj, &overflow);
```